tests/Nhs.Appointments.Api.Integration/Scenarios/UserManagement/RemoveUserRoleAssignmentsSteps.cs

```diff
@@ -0,0 +1,78 @@
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Nhs.Appointments.Api.Json;
+using Nhs.Appointments.Api.Models;
+using Xunit.Gherkin.Quick;
+using System.Text;
+using Microsoft.Azure.Cosmos;
+using Nhs.Appointments.Persistance.Models;
+
+namespace Nhs.Appointments.Api.Integration.Scenarios.UserManagement;
+
+[FeatureFile("./Scenarios/UserManagement/RemoveUserRoleAssignments.feature")]
+public sealed class RemoveUserRoleAssignmentsSteps : UserManagementBaseFeatureSteps
+{
+    private HttpResponseMessage? _response;
+    private HttpStatusCode _statusCode;
+    private RemoveUserResponse? _actualResponse;
+    
+    [When(@"I remove user '(.+)' from site '(.+)'")]
+    public async Task AssignRole(string user, string site)
+    {
+        // Check user exists before removing them for assurance this method
+        // actually works when it asserts that the removal has succeeded
+        var document = await GetUserDocument(Client, user);
+        document.Should().NotBeNull();
+
+        var userId = GetUserId(user);
+        var siteId = GetSiteId(site);
+
+        var requestBody = new RemoveUserRequest()
+        {
+            User = userId,
+            Site = siteId
+        };
+
+        _response = await Http.PostAsync($"http://localhost:7071/api/user/remove", new StringContent(JsonResponseWriter.Serialize(requestBody), Encoding.UTF8, "application/json"));
+        _statusCode = _response.StatusCode;
+        _actualResponse = await JsonRequestReader.ReadRequestAsync<RemoveUserResponse>(await _response.Content.ReadAsStreamAsync());
+    }
+    
+    [Then(@"'(.+)' does not exist in the database")]
+    public async Task Assert(string user)
+    {
+        var document = await GetUserDocument(Client, user);
+
+        document.Should().BeNull();
+    }
+
+    [Then(@"user '(.+)' would have the following role assignments")]
+    public async Task Assert(string user, Gherkin.Ast.DataTable dataTable)
+    {
+        _response.StatusCode.Should().Be(System.Net.HttpStatusCode.OK);
+
+        var userId = GetUserId(user);
+        var expectedRoleAssignments = dataTable.Rows.Skip(1).Select(
+            (row) => new RoleAssignment
+            {
+                Scope = $"site:{GetSiteId(row.Cells.ElementAt(0).Value)}",
+                Role = row.Cells.ElementAt(1).Value
+            });
+        var actualResult = await Client.GetContainer("appts", "index_data").ReadItemAsync<UserDocument>(userId, new PartitionKey("user"));
+        actualResult.Resource.RoleAssignments.Should().BeEquivalentTo(expectedRoleAssignments);
+    }
+
+    private async Task<UserDocument?> GetUserDocument(CosmosClient cosmosClient, string user)
+    {
+        var container = cosmosClient.GetContainer("appts", "index_data");
+        var document = container.GetItemLinqQueryable<UserDocument>(true)
+            .Where(d => d.Id == GetUserId(user))
+            .AsEnumerable()
+            .FirstOrDefault();
```

Comment:
This is a really weird to read a document, but `ReadItemAsync` will throw an exception if the document isn't found. This method is the best fit I could find for actually testing if something exists without excepting. Would welcome other suggestions if anyone knows a nicer way

Reply:
See if [ReadItemStreamAsync](https://learn.microsoft.com/en-us/dotnet/api/microsoft.azure.cosmos.container.readitemstreamasync?view=azure-dotnet) will work for what you need

This is in the docs for that method:
`The Stream operation only throws on client side exceptions. This is to increase performance and prevent the overhead of throwing exceptions. Check the HTTP status code on the response to check if the operation failed.`

One of the ReadItem methods would be preferable due to the RUs used for point reads vs queries although this is likely to be a lesser used function.